### PR TITLE
kafka 0.9.0.0

### PR DESCRIPTION
**Pull request by iostat**
Resolves #47978 and #48202

### CODE REVIEW

File: Library/Formula/kafka.rb [hunk 1]
```diff
@@ -63,26 +62,43 @@ def install
...
+    # namely "java.class.path", which gets set by kafka-run-class.sh, which is called by kafka-server-start.sh
+    # Thus, a correct java.class.path implies that the correct version of Kafka has been installed and is running when
+    # the command to start are invoked and the fact that it runs to begin with indicates that the
+    # prerequisites for successful environment to run in are met.
+    #
+    # We simply check that the classpath has the JAR of the Kafka version we want to use. But first, we need to
+    # set a few environment variables that kafka-run-class.sh uses for things like figuring out where
+    # to save logs. We need to do this so that Kafka doesnt try to write log files into
+    # /usr/local/Cellar/kafka/version/blahblah and instead into the sandbox's temporary directory, because otherwise
+    # otherwise instead of lovely ZooKeeper client debug messages, we just get a bunch of "operation not permitted" IO
```
Comment (miketheman) [inline Library/Formula/kafka.rb]: nitpick: "otherwise otherwise"

**Comment by akauppi** [comment]
Would like to have Kafka 0.9.0.0 on my system, via Homebrew. Is there a way I can try out this one recipe, before it gets merged to Homebrew officially?

**Comment by iostat** [comment]
in the meantime you can `brew unlink kafka` then 
`wget 'https://raw.githubusercontent.com/iostat/homebrew2/master/Library/Formula/kafka.rb' && brew install ./kafka.rb`

**Comment by bfontaine** [comment]
You can even do it with one command :wink: 

`brew install https://raw.githubusercontent.com/iostat/homebrew2/master/Library/Formula/kafka.rb`

**Comment by iostat** [comment]
TIL! :+1:

**Comment by akauppi** [comment]
Just letting you know, caveat instruction was slightly wrong. PR for @iostat here https://github.com/iostat/homebrew2/pull/1

**Comment by akauppi** [comment]
One idea for future Kafka packaging: the client side tools, `kafka-console-consumer.sh` (and 14 others) could be a separate package. It seems I'm ending in using Kafka within a Docker container, but the client tools are useful to extercise that server. 

I don't mind installing Kafka unnecessarily on the host system as well (I won't start the server), but package-wise it seems like the server and client components are two different things. Just an idea.

**Comment by bfontaine** [comment]
Sorry I didn’t mean to lock this PR.

**Comment by iostat** [comment]
@akauppi re: client side scripts: the issue with splitting them is that they all ultimately boil down to `kafka-run-class.sh`, which just runs a class that's within kafka's jars that are loaded into the classpath. The Kafka distribution you can download off of Apache's site contains both the client artifact as well as the server, and it's convenient to just extract that to an install directory where `kafka-run-class` can build up the classpath and be used for both the server and client side component.

But not only are the space savings you get from just extracting the client into a separate install are negligible, when you factor in needing to download all the dependencies kafka has, you'd basically double the disk space used if you have them in separate HB packages.

**Comment by akauppi** [comment]
@iostat Got it. Hadn't looked at what's inside those scripts.

**Comment by akauppi** [comment]
@iostat Noticed today at [docs.confluent.io](http://docs.confluent.io/2.0.1/clients/consumer.html) that 
>Its (Kafka client's) library has a much smaller footprint with no unneeded dependences. In particular, it does not depend on Kafka core.

**Pull request closed without merging**